diff -urN twin-0.3.7/server/HW/hw_tty.c twin-0.3.7-/server/HW/hw_tty.c
--- twin-0.3.7/server/HW/hw_tty.c	Sat Feb 10 00:41:46 2001
+++ twin-0.3.7-/server/HW/hw_tty.c	Thu Feb 22 20:08:17 2001
@@ -753,7 +753,8 @@
     setvbuf(stdOUT, NULL, _IOFBF, BUFSIZ);
 
     tty_number = 0;
-    if (tty_name && !strncmp(tty_name, "/dev/tty", 8)) {
+    if (tty_name && (!strncmp(tty_name, "/dev/tty", 8) ||
+    		     !strncmp(tty_name, "/dev/vc/", 8))) {
 	s = tty_name + 8;
 	while (*s && *s >= '0' && *s <= '9') {
 	    tty_number *= 10;
