--- twin-0.5.1/server/HW/hw_tty.c.old	2007-10-03 21:04:07.000000000 +0200
+++ twin-0.5.1/server/HW/hw_tty.c	2007-10-03 21:04:40.000000000 +0200
@@ -339,8 +339,8 @@
 	    } else if (!strncmp(arg, ",termcap", 8)) {
 		try_termcap = !(autotry_video = !strncmp(arg+8, "=no", 3)) << 1;
 		arg = strchr(arg+8, ',');
-	    } else if (!strncmp(arg, ",raw", 7)) {
-		try_lrawkbd = !(autotry_kbd = !strncmp(arg+7, "=no", 3)) << 1;
+	    } else if (!strncmp(arg, ",raw", 4)) {
+		try_lrawkbd = !(autotry_kbd = !strncmp(arg+4, "=no", 3)) << 1;
 		arg = strchr(arg+7, ',');
 	    } else if (!strncmp(arg, ",ctty", 5)) {
 		arg = strchr(arg+5, ',');
