diff -urN twin-0.3.7.org/MakeRules twin-0.3.7/MakeRules
--- twin-0.3.7.org/MakeRules	Thu Feb 22 19:08:27 2001
+++ twin-0.3.7/MakeRules	Thu Feb 22 19:16:44 2001
@@ -3,7 +3,18 @@
 #
 
 # set install directory
-DESTDIR=/usr/local
+
+sysconfdir = /etc
+prefix = /usr
+bindir = $(prefix)/bin
+libdir = $(prefix)/lib
+datadir = $(prefix)/share
+mandir = $(datadir)/man
+includedir = $(prefix)/include
+
+DESTDIR=
+
+INSTALL=install
 
 # choose your favorite C compiler
 CC=gcc
diff -urN twin-0.3.7.org/Makefile twin-0.3.7/Makefile
--- twin-0.3.7.org/Makefile	Thu Feb 22 19:08:27 2001
+++ twin-0.3.7/Makefile	Thu Feb 22 19:08:41 2001
@@ -22,10 +22,11 @@
 
 ifneq ($(CONF_SOCKET),n)
   install-lib:
-	$(MAKE) install -C lib
-	$(CP) include/libTw*.h $(DESTDIR)/include
+	$(MAKE) install -C lib DESTDIR=$(DESTDIR)
+	$(INSTALL) -d $(DESTDIR)$(includedir)
+	$(INSTALL) include/libTw*.h $(DESTDIR)$(includedir)
   install-clients:
-	$(MAKE) install -C clients
+	$(MAKE) install -C clients DESTDIR=$(DESTDIR)
 else
   install-lib:
   install-clients:
@@ -39,13 +40,10 @@
 	
 #
 install: all install-lib install-clients
-	$(MAKE) install -C docs
-	$(MAKE) install -C server
-	if [ -f $(DESTDIR)/lib/twin/system.twinrc ]; then \
-	   $(CP) $(DESTDIR)/lib/twin/system.twinrc $(DESTDIR)/lib/twin/system.twinrc.bak; \
-	fi && \
-	$(CP) system.twinrc $(DESTDIR)/lib/twin && \
-	$(LN) system.twinrc $(DESTDIR)/lib/twin/.twinrc
+	$(MAKE) install -C docs DESTDIR=$(DESTDIR)
+	$(MAKE) install -C server DESTDIR=$(DESTDIR)
+	$(INSTALL) -d $(DESTDIR)$(sysconfdir)
+	$(INSTALL) system.twinrc $(DESTDIR)$(sysconfdir)
 	@echo [1m ; \
 	 echo If this is the first time you installed twin $(TWIN_VERSION_STR) ; \
 	 echo you probably need to run \"ldconfig\" now. ; \
diff -urN twin-0.3.7.org/clients/Makefile twin-0.3.7/clients/Makefile
--- twin-0.3.7.org/clients/Makefile	Thu Feb 22 19:08:27 2001
+++ twin-0.3.7/clients/Makefile	Thu Feb 22 19:08:41 2001
@@ -49,8 +49,9 @@
 Torture:
 
 install:
-	$(CP) $(BINS) */tw* $(DESTDIR)/bin 
-	$(LN) twattach $(DESTDIR)/bin/twdetach
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL) $(BINS) */tw* $(DESTDIR)$(bindir)
+	$(LN) twattach $(DESTDIR)$(bindir)/twdetach
 
 clean:
 	rm -f .*.flags .*.link gmon.out core $(OBJS) $(BINS) twdetach
diff -urN twin-0.3.7.org/contrib/Makefile twin-0.3.7/contrib/Makefile
--- twin-0.3.7.org/contrib/Makefile	Thu Feb 22 19:08:27 2001
+++ twin-0.3.7/contrib/Makefile	Thu Feb 22 19:08:41 2001
@@ -39,7 +39,8 @@
 endif
 
 install:
-	$(CP) $(BINS) $(DESTDIR)/bin 
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL) $(BINS) $(DESTDIR)$(bindir)
 
 clean:
 	rm -f .*.flags .*.link gmon.out core $(OBJS) $(BINS)
diff -urN twin-0.3.7.org/docs/Makefile twin-0.3.7/docs/Makefile
--- twin-0.3.7.org/docs/Makefile	Thu Feb 22 19:08:27 2001
+++ twin-0.3.7/docs/Makefile	Thu Feb 22 19:08:41 2001
@@ -40,10 +40,5 @@
 clean:
 
 install:
-	if [ -d $(DESTDIR)/share/doc ]; then \
-	  $(MKDIR) $(DESTDIR)/share/doc/twin/docs && \
-	  $(CP) $(DOCS) $(DESTDIR)/share/doc/twin/docs; \
-	  cd $(TOPDIR) && \
-	  $(CP) $(TOPDOCS) $(DESTDIR)/share/doc/twin; \
-	fi
-
+	$(INSTALL) -d $(DESTDIR)$(mandir)/man1
+	$(INSTALL) twin.1 $(DESTDIR)$(mandir)/man1
diff -urN twin-0.3.7.org/lib/Makefile twin-0.3.7/lib/Makefile
--- twin-0.3.7.org/lib/Makefile	Thu Feb 22 19:08:27 2001
+++ twin-0.3.7/lib/Makefile	Thu Feb 22 19:08:41 2001
@@ -3,6 +3,18 @@
 #
 TOPDIR=..
 
+sysconfdir = /etc
+prefix = /usr
+bindir = $(prefix)/bin
+libdir = $(prefix)/lib
+datadir = $(prefix)/share
+mandir = $(datadir)/man
+includedir = $(prefix)/include
+
+DESTDIR=
+
+INSTALL=install
+
 all: everything
 
 VERSION=2.0.4
@@ -77,12 +89,13 @@
 Torture: torture
 
 install-real:
+	$(INSTALL) -d $(DESTDIR)$(libdir)
 	if [ -f libTw.so.$(VERSION) ]; then \
-	  $(CP) libTw.so.$(VERSION) $(DESTDIR)/lib ; \
-	  $(LN) libTw.so.$(VERSION) $(DESTDIR)/lib/libTw.so.$(SO_VER) ; \
-	  $(LN) libTw.so.$(SO_VER)  $(DESTDIR)/lib/libTw.so ; \
+		$(INSTALL) libTw.so.$(VERSION) $(DESTDIR)$(libdir); \
+		$(LN) libTw.so.$(VERSION) $(DESTDIR)$(libdir)/libTw.so.$(SO_VER); \
+		$(LN) libTw.so.$(SO_VER)  $(DESTDIR)$(libdir)/libTw.so; \
 	else \
-	  $(CP) libTw.a $(DESTDIR)/lib ; \
+		$(INSTALL) libTw.a $(DESTDIR)$(libdir); \
 	fi
 	
 clean:
diff -urN twin-0.3.7.org/server/HW/Makefile twin-0.3.7/server/HW/Makefile
--- twin-0.3.7.org/server/HW/Makefile	Thu Feb 22 19:08:27 2001
+++ twin-0.3.7/server/HW/Makefile	Thu Feb 22 19:08:41 2001
@@ -141,7 +141,7 @@
 
 install:
 	if [ "$(MODBINS)" ]; then \
-	  $(MKDIR) $(DESTDIR)/lib/twin/modules/HW; \
-	  $(CP) $(MODBINS) $(DESTDIR)/lib/twin/modules/HW; \
-	  $(LN) hw_X11.so $(DESTDIR)/lib/twin/modules/HW/hw_X.so; \
+	  $(INSTALL) -d $(DESTDIR)$(libdir)/twin/modules/HW; \
+	  $(INSTALL) $(MODBINS) $(DESTDIR)$(libdir)/twin/modules/HW; \
+	  $(LN) hw_X11.so $(DESTDIR)$(libdir)/twin/modules/HW/hw_X.so; \
 	fi
diff -urN twin-0.3.7.org/server/Makefile twin-0.3.7/server/Makefile
--- twin-0.3.7.org/server/Makefile	Thu Feb 22 19:08:27 2001
+++ twin-0.3.7/server/Makefile	Thu Feb 22 19:19:03 2001
@@ -5,6 +5,10 @@
 
 TOPDIR=..
 
+prefix = /usr
+bindir = $(prefix)/bin
+libdir = $(prefix)/lib
+
 BASEOBJS=data.o builtin.o methods.o util.o draw.o extensions.o \
 	main.o hw_multi.o printk.o resize.o scroller.o remote.o alloc.o
 HWOBJS=hw.o HW/hw.a
@@ -29,10 +33,9 @@
 BINS=twin_real twin_wrapper
 MODBINS=
 
-# don't use := here, as DESTDIR is not set yet
-CFLAGS_data.o=$(strip $(patsubst %,-DDESTDIR=\"%\",$(DESTDIR)))
-CFLAGS_wrapper.o=$(strip $(patsubst %,-DDESTDIR=\"%\",$(DESTDIR)))
-CFLAGS_display.o=$(strip $(patsubst %,-DDESTDIR=\"%\",$(DESTDIR)))
+CFLAGS_data.o=-DBINDIR=\"$(bindir)\" -DLIBDIR=\"$(libdir)\"
+CFLAGS_wrapper.o=-DBINDIR=\"$(bindir)\" -DLIBDIR=\"$(libdir)\"
+CFLAGS_display.o=-DBINDIR=\"$(bindir)\" -DLIBDIR=\"$(libdir)\"
 
 LDFLAGS_wm.so=-lc -shared
 LDFLAGS_rcparse.so=-lc -shared
@@ -236,13 +239,13 @@
 	rm -f rcparse_tab.c rcparse_tab.h rcparse_lex.c
 
 install:
-	$(CP) $(BINS) twstart $(DESTDIR)/bin
-	$(LN) twin_wrapper $(DESTDIR)/bin/twin
+	$(INSTALL) -d $(DESTDIR)$(bindir) $(DESTDIR)/$(libdir)/twin/modules
+	$(INSTALL) $(BINS) twstart $(DESTDIR)$(bindir)
+	$(LN) twin_wrapper $(DESTDIR)$(bindir)/twin
 	if [ "$(MODBINS)" ]; then \
-	  $(MKDIR) $(DESTDIR)/lib/twin/modules; \
-	  $(CP) $(MODBINS) $(DESTDIR)/lib/twin/modules; \
+	  $(INSTALL) $(MODBINS) $(DESTDIR)$(libdir)/twin/modules; \
 	fi
-	$(MAKE) install -C HW
+	$(MAKE) install -C HW DESTDIR=$(DESTDIR)
 	@if [ "$(SETPRIV)" ]; then \
 	  echo [1m ; \
 	  echo Twin may need special privileges for the terminal emulator to work. ; \
@@ -253,4 +256,3 @@
 	  echo you will grant special privileges to twin AT YOUR OWN RISK. ; \
 	  echo [0m ; \
 	fi
-
diff -urN twin-0.3.7.org/server/data.c twin-0.3.7/server/data.c
--- twin-0.3.7.org/server/data.c	Thu Feb 22 19:08:27 2001
+++ twin-0.3.7/server/data.c	Thu Feb 22 19:09:28 2001
@@ -20,9 +20,9 @@
 
 /* setup configuration paths */
 
-#ifdef DESTDIR
-CONST byte *conf_destdir_lib_twin = DESTDIR "/lib/twin";
-CONST byte *conf_destdir_lib_twin_modules_ = DESTDIR "/lib/twin/modules/";
+#ifdef LIBDIR
+CONST byte *conf_destdir_lib_twin = LIBDIR "/twin";
+CONST byte *conf_destdir_lib_twin_modules_ = LIBDIR "/twin/modules/";
 #else
 CONST byte *conf_destdir_lib_twin = ".";
 CONST byte *conf_destdir_lib_twin_modules_ = "./";
diff -urN twin-0.3.7.org/server/display.c twin-0.3.7/server/display.c
--- twin-0.3.7.org/server/display.c	Thu Feb 22 19:08:27 2001
+++ twin-0.3.7/server/display.c	Thu Feb 22 19:09:48 2001
@@ -59,8 +59,8 @@
 
 #ifdef CONF__MODULES
 
-# ifdef DESTDIR
-#  define DIR_LIB_TWIN_MODULES_ DESTDIR "/lib/twin/modules/"
+# ifdef LIBDIR
+#  define DIR_LIB_TWIN_MODULES_ LIBDIR "/twin/modules/"
 # else
 #  define DIR_LIB_TWIN_MODULES_ "./"
 # endif
diff -urN twin-0.3.7.org/server/wrapper.c twin-0.3.7/server/wrapper.c
--- twin-0.3.7.org/server/wrapper.c	Thu Feb 22 19:08:27 2001
+++ twin-0.3.7/server/wrapper.c	Thu Feb 22 19:10:10 2001
@@ -11,8 +11,8 @@
 #include <string.h>
 
 int main(int argc, char *argv[]) {
-#ifdef DESTDIR
-    argv[0] = DESTDIR "/bin/twin_real";
+#ifdef BINDIR
+    argv[0] = BINDIR "/twin_real";
 #else
     argv[0] = "./twin_real";
 #endif
