diff -Nru twin-0.3.7/MakeRules twin-0.3.7.new/MakeRules
--- twin-0.3.7/MakeRules	Thu Feb 22 06:24:36 2001
+++ twin-0.3.7.new/MakeRules	Thu Feb 22 06:25:56 2001
@@ -31,10 +31,12 @@
 # compile/link flags. Always use += to set variables here!
 #
 ifdef DEBUG
- CFLAGS+=-g -DINLINE=static -pipe
+ OPT_FLAGS=-g
+ CFLAGS+=$(OPT_FLAGS) -DINLINE=static -pipe
  LDFLAGS+=-g
 else
- CFLAGS+=-O2 -fomit-frame-pointer -DINLINE="extern inline" -pipe
+ OPT_FLAGS=-O2 -fomit-frame-pointer
+ CFLAGS+=$(OPT_FLAGS) -DINLINE="extern inline" -pipe
  LDFLAGS+=-s
 endif
 
