--- twin-0.8.1/configure.ac.orig	2023-02-05 19:51:46.010582238 +0100
+++ twin-0.8.1/configure.ac	2023-02-05 20:03:31.693425892 +0100
@@ -86,7 +86,7 @@ for try_flag in -fomit-frame-pointer -fn
 do
   save_CFLAGS="$CFLAGS"
   CFLAGS="$CFLAGS $try_flag"
-  try_flag_id="`echo \"$try_flag\" | tr - _`"
+  try_flag_id="$(echo "$try_flag" | tr - _)"
   AC_CACHE_CHECK(for C compiler flag $try_flag, ac_cv_cflag$try_flag_id, AC_TRY_COMPILE(
   [
     int ret0(void) {
@@ -269,7 +269,7 @@ TW_FIX_CONFIG
 
 
 if test "$enable__dlopen" = yes ; then
-  AC_SUBST(LIBDL, "`echo \"$ac_cv_search_dlopen\"         | $SED -e 's/^none required$//g' -e 's/^no$//g'`")
+  AC_SUBST(LIBDL, "$(echo "$ac_cv_search_dlopen"         | $SED -e 's/^none required$//g' -e 's/^no$//g')")
 else
   # if dlopen() or <dlfcn.h> are missing, or user specified --enable-dlopen=no, link against libtool
   AC_SUBST(LIBDL, "$LIBLTDL")
@@ -280,16 +280,16 @@ else
   fi
 fi
 
-AC_SUBST(LIBCRYPT,    "`echo \"$ac_cv_search_crypt\"          | $SED -e 's/^none required$//g' -e 's/^no$//g'`")
-AC_SUBST(LIBGGI,      "`echo \"$ac_cv_search_ggiOpen\"        | $SED -e 's/^none required$//g' -e 's/^no$//g'`")
-AC_SUBST(LIBGPM,      "`echo \"$ac_cv_search_Gpm_Open\"       | $SED -e 's/^none required$//g' -e 's/^no$//g'`")
-AC_SUBST(LIBSOCK,     "`echo \"$ac_cv_search_gethostbyname\"  | $SED -e 's/^none required$//g' -e 's/^no$//g'` `echo \"$ac_cv_search_socket\" | $SED -e 's/^none required$//g' -e 's/^no$//g'` `echo \"$ac_cv_search_connect\"  | $SED -e 's/^none required$//g' -e 's/^no$//g'`")
-AC_SUBST(LIBPTHREAD,  "`echo \"$ac_cv_search_pthread_create\" | $SED -e 's/^none required$//g' -e 's/^no$//g'`")
-AC_SUBST(LIBTERMCAP,  "`echo \"$ac_cv_search_tgetent\"        | $SED -e 's/^none required$//g' -e 's/^no$//g'`")
-AC_SUBST(LIBXPM,      "`echo \"$ac_cv_search_XpmReadFileToPixmap\"  | $SED -e 's/^none required$//g' -e 's/^no$//g'`")
-AC_SUBST(LIBXFT,      "`echo \"$ac_cv_search_XftFontOpen\"    | $SED -e 's/^none required$//g' -e 's/^no$//g'`")
+AC_SUBST(LIBCRYPT,    "$(echo "$ac_cv_search_crypt"          | $SED -e 's/^none required$//g' -e 's/^no$//g')")
+AC_SUBST(LIBGGI,      "$(echo "$ac_cv_search_ggiOpen"        | $SED -e 's/^none required$//g' -e 's/^no$//g')")
+AC_SUBST(LIBGPM,      "$(echo "$ac_cv_search_Gpm_Open"       | $SED -e 's/^none required$//g' -e 's/^no$//g')")
+AC_SUBST(LIBSOCK,     "$(echo "$ac_cv_search_gethostbyname"  | $SED -e 's/^none required$//g' -e 's/^no$//g') $(echo "$ac_cv_search_socket" | $SED -e 's/^none required$//g' -e 's/^no$//g') $(echo "$ac_cv_search_connect"  | $SED -e 's/^none required$//g' -e 's/^no$//g')")
+AC_SUBST(LIBPTHREAD,  "$(echo "$ac_cv_search_pthread_create" | $SED -e 's/^none required$//g' -e 's/^no$//g')")
+AC_SUBST(LIBTERMCAP,  "$(echo "$ac_cv_search_tgetent"        | $SED -e 's/^none required$//g' -e 's/^no$//g')")
+AC_SUBST(LIBXPM,      "$(echo "$ac_cv_search_XpmReadFileToPixmap"  | $SED -e 's/^none required$//g' -e 's/^no$//g')")
+AC_SUBST(LIBXFT,      "$(echo "$ac_cv_search_XftFontOpen"    | $SED -e 's/^none required$//g' -e 's/^no$//g')")
 if test "$enable_socket_gz" = yes; then
-  AC_SUBST(LIBZ,      "`echo \"$ac_cv_search_deflate\"        | $SED -e 's/^none required$//g' -e 's/^no$//g'`")
+  AC_SUBST(LIBZ,      "$(echo "$ac_cv_search_deflate"        | $SED -e 's/^none required$//g' -e 's/^no$//g')")
 fi
 AC_SUBST(LIBTUTF,    '$(top_builddir)/libs/libTutf/libTutf.la')
 AC_SUBST(LIBTW,      '$(top_builddir)/libs/libTw/libTw.la')
